### PR TITLE
Use real32 floats in WaveSolver

### PR DESCRIPTION
**Pull request by XL64**
This PR change real64 to real32 in waveSolvers

Rebaseline done in https://github.com/GEOSX/integratedTests/pull/266

### CODE REVIEW

**Review by acitrain — APPROVED** [review]

File: src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.hpp [hunk 1]
```diff
@@ -203,63 +203,63 @@ namespace extrinsicMeshData
 
 EXTRINSIC_MESH_DATA_TRAIT( Pressure_nm1,
                            "pressure_nm1",
-                           array1d< real64 >,
+                           real32_array,
```
Comment (sframba) [inline src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.hpp:206]: Question: why `real32_array` and not `array1d< real32 >`?

Comment (XL64) [reply src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.hpp:206]: There is no particular reason, they are aliases. I could have used either, do you want me to change it ?

Comment (TotoGaz) [reply src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.hpp:206]: Yes please, we tend not to use those aliases too much.

**Review by sframba — APPROVED** [review]

**Comment by rrsettgast** [comment]
Are you always going to want `real32` for these variables? i.e. Are there cases where you would want to keep `real64`? If so, perhaps it makes sense to template `AcousticWaveEquationSEM` on the floating point type so you can easily have both?

**Comment by XL64** [comment]
> 

We should not need real64 precision.

**Comment by sframba** [comment]
image
Comparison between real32 and real32. The results are identical within numerical precision.

**Pull request merged**